pipelines/check_records.py

```
@@ -264,6 +287,49 @@ def root_record_id(obj):
+    errs = []
+
+    def walk(node, path):
+        if isinstance(node, dict):
+            for k, v in node.items():
+                cur = f"{path}.{k}" if path else k
+                if k == "sim_or_real" and isinstance(v, str):
+                    if "real" in v.lower() or v not in ALLOWED_PROVENANCE:
+                        # 'real' substring is always an error; other invalid values already
+                        # produce a warning via expected_states but also flag here if
+                        # they contain 'real'. Keep error strictly for 'real'.
+                        if "real" in v.lower():
```

Ruff (0.16.1)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)